internal/app/components/timelines/timelines.go

```diff
@@ -0,0 +1,71 @@
+package timelines
+
+import (
+	"github.com/gzcharleszhang/course-planner/internal/app/components/courses"
+	"github.com/gzcharleszhang/course-planner/internal/app/components/plans"
+	"github.com/gzcharleszhang/course-planner/internal/app/components/terms"
+	"github.com/rs/xid"
+)
+
+type TimelineId string
+type TimelineName string
+
+type Timeline struct {
+	Id          TimelineId
+	Name        TimelineName
+	TermRecords []*terms.TermRecord
+	Plans       plans.Plans
+}
+
+func newTimelineId() TimelineId {
+	return TimelineId(xid.New().String())
+}
+
+func NewTimeline(name TimelineName) *Timeline {
```

Comment:
can you add a todo here for using course history when adding a new timeline? this method could prob just take `CourseRecords` as an additional parameter

Reply:
just a TODO is fine for this PR tho